core/src/util/usLog_p.h

```diff
@@ -69,25 +69,82 @@ struct LogMsg {
   std::stringstream buffer;
 };
 
+class Logger {
+public:
+    static Logger& instance()
+    { 
+      /* 
+      IMPORTANT: This is only thread safe if compiling with a C++11 compiler
+        which implements C++11 standard section 6.7.4:
+       "If control enters the declaration concurrently while the variable is being initialized, 
+        the concurrent execution shall wait for completion of the initialization."
+
+        However, even if a C++11 supported compiler isn't used, thread safe initialization
```

Comment:
I mention it because thread-safety will become an issue if the LogMsg interface is made public, as shown in [branch 25-module-lifecycle-enhancements](https://github.com/saschazelzer/CppMicroServices/tree/25-module-lifecycle-enhancements) in [this commit](https://github.com/saschazelzer/CppMicroServices/commit/b3f275f44641166a9618a99c6178b1bb722c5090)